models/addresses_events/avalanche_c/addresses_events_avalanche_c_first_funded_by.sql

```diff
@@ -0,0 +1,44 @@
+{{ config(
+    schema = 'addresses_events_avalanche_c'
+    , alias = 'first_funded_by'
+    , materialized = 'incremental'
+    , file_format = 'delta'
+    , incremental_strategy = 'merge'
+    , unique_key = ['address']
+    )
+}}
+
+SELECT 'avalanche_c' AS blokchain
+    , b.to AS address
+    , MIN(a.from) AS first_funded_by
+    , MIN(a.block_time) AS block_time
+    , MIN(a.block_number) AS block_number
+    , MIN(a.tx_hash) AS tx_hash
+FROM {{ source('avalanche_c', 'traces') }} a
+JOIN
+(
+    SELECT et.to
+        , MIN(et.block_number) AS first_block
+    FROM {{ source('avalanche_c', 'traces') }} et
+    {% if is_incremental() %}
+    LEFT ANTI JOIN {{this}} ffb
+        ON et.to = ffb.address
+    {% endif %}
+    WHERE et.success
+    AND (et.call_type NOT IN ('delegatecall', 'callcode', 'staticcall') OR et.call_type IS NULL)
+    AND et.value > 0
+    {% if is_incremental() %}
+    AND et.block_time >= date_trunc('day', now() - interval '1 week')
+    {% endif %}
+    GROUP BY et.to
+) AS b 
+ON a.to = b.to
+    AND a.block_number = b.first_block
+WHERE a.success
+    AND (a.call_type NOT IN ('delegatecall', 'callcode', 'staticcall') OR a.call_type IS NULL)
+    AND a.value > 0
+    {% if is_incremental() %}
+    AND a.block_time >= date_trunc('day', now() - interval '1 week')
+    {% endif %}
+GROUP BY b.to
```

Comment:
Can we try the `MIN_BY` setup for these queries?
Something like this (haven't tested)
```
SELECT latest.* 
FROM (
  SELECT 'ethereum' AS blokchain
      , et.to AS address
      , MIN_BY(et.from, et.block_number) AS first_funded_by
      , MIN(et.block_time) AS block_time
      , MIN(et.block_number) AS block_number
      , MIN_BY(et.tx_hash, et.block_number) AS tx_hash
  FROM {{ source('ethereum', 'traces') }} et
  WHERE et.success
  AND (et.call_type NOT IN ('delegatecall', 'callcode', 'staticcall') OR et.call_type IS NULL)
  AND et.value > 0
  {% if is_incremental() %}
  AND et.block_time >= date_trunc('day', now() - interval '1 week')
  {% endif %}
  GROUP BY et.to
  ) latest
{% if is_incremental() %}
LEFT ANTI JOIN {{this}} ffb
    ON latest.address = ffb.address
{% endif %}


```

Reply:
I didnt know about the MIN_BY() function, so cool! I'll try it out, it definitely should work